native/spark-expr/src/utils.rs

```diff
@@ -146,6 +162,25 @@ fn timestamp_ntz_to_timestamp(
             };
             Ok(Arc::new(array_with_tz))
         }
+        DataType::Timestamp(TimeUnit::Millisecond, None) => {
+            let array = as_primitive_array::<TimestampMillisecondType>(&array);
+            let tz: Tz = tz.parse()?;
```

Comment:
Is this called frequently (per row)?  timezone parse is somewhat expensive (and does not change for a session).

Reply:
This is once per array, but I think the parsing could happen once during planning rather than per batch/array.

Reply:
Makes sense, we can defer this for the moment.